server-spi-private/src/main/java/org/keycloak/userprofile/DefaultAttributeGroups.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.userprofile;
+
+import org.keycloak.models.Constants;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.validate.ValidationContext;
+import org.keycloak.validate.ValidationError;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * <p>The default implementation for {@link AttributeGroups}.
+ * Should be reused as much as possible by the different implementations of {@link UserProfileProvider}.
+ *
+ * <p>This implementation is not specific to any user profile implementation.
+ *
+ * @author <a href="mailto:joerg.matysiak@bosch.io">Jörg Matysiak</a>
+ */
+public class DefaultAttributeGroups implements AttributeGroups {
```

Comment:
We had a discussion internally whether this class should be better named `AttributeGroupsImpl` - if so the rename should probably also been applied to `DefaultAttributes`. @pedroigor  @velias  what do you think?

Reply:
I'm not sure if we need a specific hierarchy/abstraction for groups. We should be fine by just leveraging the `Attributes` interface. 

It should make things more simple.